sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -911,9 +911,9 @@ public boolean requiresDeduping() {
     @VisibleForTesting
     static final String METRIC_NAMESPACE = "KafkaIOReader";
     @VisibleForTesting
-    static final String CHECKPOINT_MARK_COMMITS_METRIC = "checkpointMarkCommits";
-    private static final String CHECKPOINT_MARK_COMMIT_SKIPS_METRIC = "checkpointMarkCommitSkips";
-
+    static final String CHECKPOINT_MARK_COMMITS_ENQUEUED_METRIC = "checkpointMarkCommitsEnqueued";
+    private static final String CHECKPOINT_MARK_COMMITS_SKIPPED_METRIC =
```

Comment:
Can you have more description about the definition of the two metrics? I'm a little confused that they're only related with `KafkaCheckpointMark.finalizeCheckpoint()`, but not method `commitCheckpointMark(KafkaCheckpointMark checkpointMark)` who commit the offset.

Reply:
We only keep one pending commit at most. When a new finalizeCheckpoint() request arrives, if the old one is still not processed, it is dropped in favor of the newer one. 
We have two counters : `commitsEnqueued` & `commitsSkipped`. Number of actual commits = `commitsEnqueued - commitsSkipped` (this assumes there is nothing being processed or queued. so could be off by at most 2, though often it is accurate). 

Given that we can not update metrics directly inside `commitCheckpointMark`, this is the simplest I could think of. Alternate is to count the commits in another atomic int and update the metric inside `advance()` or other UnboundedSource API.

Reply:
Thanks for the explanation, will merge it soon.